app/src/main/kotlin/com/wire/android/ui/home/conversations/model/messagetypes/audio/AudioMessageType.kt

```diff
@@ -0,0 +1,217 @@
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryIconButton
+import com.wire.android.ui.common.clickable
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.spacers.HorizontalSpace
+import com.wire.android.ui.theme.wireColorScheme
+
+@Composable
+fun AudioMessage(
+    audioMediaPlayingState: AudioMediaPlayingState,
+    totalTimeInMs: Int,
+    currentPositionInMs: Int,
+    onPlayButtonClick: () -> Unit,
+    onSliderPositionChange: (Float) -> Unit,
+    onAudioMessageLongClick: (() -> Unit)?
+) {
+    Box(
+        modifier = Modifier
+            .padding(top = dimensions().spacing4x)
+            .background(
+                color = MaterialTheme.wireColorScheme.onPrimary,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.wireColorScheme.secondaryButtonDisabledOutline,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .padding(dimensions().spacing8x)
+    ) {
+        if (audioMediaPlayingState is AudioMediaPlayingState.Failed) {
+            FailedAudioMessage()
+        } else {
+            SuccessFullAudioMessage(
+                audioMediaPlayingState = audioMediaPlayingState,
+                totalTimeInMs = totalTimeInMs,
+                currentPositionInMs = currentPositionInMs,
+                onPlayButtonClick = onPlayButtonClick,
+                onSliderPositionChange = onSliderPositionChange,
+                onAudioMessageLongClick = onAudioMessageLongClick
+            )
+        }
+    }
+}
+
+@Composable
+private fun SuccessFullAudioMessage(
+    audioMediaPlayingState: AudioMediaPlayingState,
+    totalTimeInMs: Int,
+    currentPositionInMs: Int,
+    onPlayButtonClick: () -> Unit,
+    onSliderPositionChange: (Float) -> Unit,
+    onAudioMessageLongClick: (() -> Unit)? = null
+) {
+    val audioDuration by remember(currentPositionInMs) {
+        mutableStateOf(
+            AudioDuration(totalTimeInMs, currentPositionInMs)
+        )
+    }
+
+    val isFetching = audioMediaPlayingState is AudioMediaPlayingState.Fetching
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(dimensions().audioMessageHeight)
+            .clickable(Clickable(true, onLongClick = onAudioMessageLongClick)),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        WireSecondaryIconButton(
+            minWidth = 32.dp,
+            iconResource = getPlayOrPauseIcon(audioMediaPlayingState),
+            shape = CircleShape,
+            contentDescription = R.string.content_description_image_message,
+            state = if (isFetching) WireButtonState.Disabled else WireButtonState.Default,
```

Comment:
I think the icon color should also be grayed out while downloading according to design specs.

Reply:
it actually is grayed out as well, but thanks for sharp eye